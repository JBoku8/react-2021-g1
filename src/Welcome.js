const Welcome = (props) => {
  const styles = {
    color: "red",
    textAlign: "center",
  };

  return (
    <div>
      <h1 className="title" style={styles}>
        {props.title}
      </h1>
    </div>
  );
};

export default Welcome;
