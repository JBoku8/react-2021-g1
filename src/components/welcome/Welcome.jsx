import { useState } from "react";
import Example from "../example";
import { Button } from "../ui";

const Welcome = ({ title, startNumber }) => {
  const [counter, setCounter] = useState(startNumber);
  const styles = {
    color: "red",
    textAlign: "center",
  };

  // newValue !== oldValue
  // newObject !== oldObject

  const onAdd = () => {
    setCounter(counter + 1);
  };
  const onMinus = () => {
    setCounter(counter - 1);
  };
  const onReset = () => {
    setCounter(startNumber);
  };

  return (
    <div>
      <h1 className="title" style={styles}>
        {title}
      </h1>
      <Example message={counter} />
      <Button onClick={onReset} text="Reset" type="default" />
      <Button onClick={onAdd} text="Add" type="danger" />
      <Button onClick={onMinus} text="Minus" type="warning" />
    </div>
  );
};

export default Welcome;
