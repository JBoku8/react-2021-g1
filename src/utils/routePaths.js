export const TODO_PATH = '/todo-list';
export const COUNTER_PATH = '/counter';
export const PROFILE_PATH = '/profile';
export const FAKER_BOOKS_PATH = '/faker-books';
export const REDUX_COUNTER_PATH = '/redux-counter';
export const HOME = '/';

export const AUTH_PATH = '/auth';
export const AUTH_SIGN_IN = '/auth/sign-in';
export const AUTH_SIGN_UP = '/auth/sign-up';
