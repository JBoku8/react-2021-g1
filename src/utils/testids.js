export const TEST_IDS = {
  example: {
    message: 'example.message',
  },
  welcome: {
    title: 'welcome.title',
    resetButton: 'welcome.resetButton',
    addButton: 'welcome.addButton',
    minusButton: 'welcome.minusButton',
  },
};
