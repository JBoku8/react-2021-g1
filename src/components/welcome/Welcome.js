import { useState } from 'react';
import { TEST_IDS } from '../../utils/testids';
import Example from '../example';
import Button from '../ui/Button';

const Welcome = ({ title, startNumber }) => {
  const [counter, setCounter] = useState(startNumber);
  const styles = {
    color: 'red',
    textAlign: 'center',
  };

  // newValue !== oldValue
  // newObject !== oldObject

  const onAdd = () => {
    setCounter(counter + 1);
  };
  const onMinus = () => {
    setCounter(counter - 1);
  };
  const onReset = () => {
    setCounter(startNumber);
  };

  return (
    <div className="row">
      <h1 className="title" style={styles} data-testid={TEST_IDS.welcome.title}>
        {title}
      </h1>
      <Example message={counter} />
      <Button
        onClick={onReset}
        text="Reset"
        type="default"
        data-testid={TEST_IDS.welcome.resetButton}
      />
      <Button onClick={onAdd} text="Add" type="danger" data-testid={TEST_IDS.welcome.addButton} />
      <Button
        onClick={onMinus}
        text="Minus"
        type="warning"
        data-testid={TEST_IDS.welcome.minusButton}
      />
    </div>
  );
};

export default Welcome;
