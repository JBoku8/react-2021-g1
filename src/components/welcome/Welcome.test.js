import { render } from '@testing-library/react';
import { TEST_IDS } from '../../utils/testids';
import Welcome from './Welcome';

const mockProps = {
  title: 'mock-title',
  startNumber: 0,
};

describe('Welcome component testing', () => {
  let component;
  beforeEach(() => {
    component = render(<Welcome {...mockProps} />);
  });

  it('should render title correctly', () => {
    const el = component.getByTestId(TEST_IDS.welcome.title);
    expect(el).toHaveTextContent(mockProps.title);
  });

  it('should increase the startNumber by 1', () => {
    const message = component.getByTestId(TEST_IDS.example.message);
    const addButton = component.getByTestId(TEST_IDS.welcome.addButton);

    addButton.click();

    expect(message).toHaveTextContent(mockProps.startNumber + 1);
  });

  it('should decrease the startNumber by 1', () => {
    const message = component.getByTestId(TEST_IDS.example.message);
    const minusButton = component.getByTestId(TEST_IDS.welcome.minusButton);

    minusButton.click();

    expect(message).toHaveTextContent(mockProps.startNumber - 1);
  });

  it('should reset the startNumber by to defaults', () => {
    const message = component.getByTestId(TEST_IDS.example.message);
    const minusButton = component.getByTestId(TEST_IDS.welcome.minusButton);
    const resetButton = component.getByTestId(TEST_IDS.welcome.resetButton);
    minusButton.click();
    expect(message).toHaveTextContent(mockProps.startNumber - 1);

    resetButton.click();
    expect(message).toHaveTextContent(mockProps.startNumber);
  });
});

describe('Welcome snapshot testing', () => {
  test('welcome renders as expected', () => {
    const component = render(<Welcome {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
});
