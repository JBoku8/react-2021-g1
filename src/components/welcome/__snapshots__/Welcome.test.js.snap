// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome snapshot testing welcome renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <h1
          class="title"
          data-testid="welcome.title"
          style="color: red; text-align: center;"
        >
          mock-title
        </h1>
        <div>
          <h1
            class="text-muted fs-1"
            data-testid="example.message"
          >
            0
          </h1>
        </div>
        <button
          class="btn default mx-1"
          data-testid="welcome.resetButton"
          type="default"
        >
          Reset
        </button>
        <button
          class="btn danger mx-1"
          data-testid="welcome.addButton"
          type="danger"
        >
          Add
        </button>
        <button
          class="btn warning mx-1"
          data-testid="welcome.minusButton"
          type="warning"
        >
          Minus
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <h1
        class="title"
        data-testid="welcome.title"
        style="color: red; text-align: center;"
      >
        mock-title
      </h1>
      <div>
        <h1
          class="text-muted fs-1"
          data-testid="example.message"
        >
          0
        </h1>
      </div>
      <button
        class="btn default mx-1"
        data-testid="welcome.resetButton"
        type="default"
      >
        Reset
      </button>
      <button
        class="btn danger mx-1"
        data-testid="welcome.addButton"
        type="danger"
      >
        Add
      </button>
      <button
        class="btn warning mx-1"
        data-testid="welcome.minusButton"
        type="warning"
      >
        Minus
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
